### PR TITLE
Remove generics from GroupStateStorage

### PR DESCRIPTION
**Pull request by mulmarta**
Make GroupStateStorage operate on bytes instead of mls-encodable generics.

The risk of confusing bytes with other bytes is mitigated by using `GroupState` and `EopchRecord` structs.

~~### Call-outs~~

~~`GroupState` and `EopchRecord` derive UniFFI export if the "uniffi" feature is used, but they're not used yet due to an issue with Python tests. I'm not sure how we feel about using UniFFI outside of the mls-rs-uniffi crate? I don't have strong opinions.~~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT license.

### CODE REVIEW

File: mls-rs-core/src/group/group_state.rs [hunk 1]
```diff
@@ -2,22 +2,35 @@
...
+
 use crate::error::IntoAnyError;
 #[cfg(mls_build_async)]
 use alloc::boxed::Box;
 use alloc::vec::Vec;
-use mls_rs_codec::{MlsDecode, MlsEncode};
 
 /// Generic representation of a group's state.
-pub trait GroupState {
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
```
[the remark this answers is not in the document]
Comment (stefunctional) [reply mls-rs-core/src/group/group_state.rs:13]: Sorry I didn't make it clear in my pretty-printing PR. To affect how byte arrays are pretty-printed for `Debug`, a manual implementation is needed. This manual implementation can call `mls_rs_core::debug::pretty_bytes` to print only the length by default, unless `"{:#?}"` is used, which also causes the bytes to be output. Does that make sense?

[the remark this answers is not in the document]
Comment (mulmarta) [reply mls-rs-core/src/group/group_state.rs]: Forgot about that, sorry! Fixed.

File: mls-rs-uniffi/src/config.rs [hunk 2]
```diff
@@ -37,7 +53,49 @@ pub type UniFFIConfig = client_builder::WithIdentityProvider<
...
+
+#[cfg(feature = "sqlite")]
+#[uniffi::export]
+impl ClientConfig {
+    #[uniffi::constructor]
+    pub fn new(group_state_storage: Arc<dyn GroupStateStorage>) -> Self {
+        Self {
+            group_state_storage,
+        }
+    }
```
Comment (stefunctional) [inline mls-rs-uniffi/src/config.rs:80]: This function looks identical to the one in the block above when `sqlite` is off. Is there some `uniffi` limitation that forces us to duplicate it?

File: mls-rs-core/src/group/group_state.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (stefunctional) [reply mls-rs-core/src/group/group_state.rs]: No worries! Thank you!

**Review by stefunctional — APPROVED** [review]

File: mls-rs-uniffi/src/config.rs (hunk 2 above)
Comment (mulmarta) [reply mls-rs-uniffi/src/config.rs]: Sorry for mixing unrelated changes by mistake. I reverted the commit.

Comment (stefunctional) [reply mls-rs-uniffi/src/config.rs]: No problem. I was wondering why sqlite-related changes made it in this PR but I thought there must have been some reason I was not aware of :p

**Review by tomleavy — APPROVED** [review]

**Pull request merged**